🐞 [Bug Report] - Playwright dependency errors

**Describe the bug**
Mainline builds are currently failing with playwright errors before UI tests start

**To Reproduce**
1. Try and build
1. Receive failure

#271 
#272 
#276 

```

BEWARE: your OS is not officially supported by Playwright; downloading fallback build.
...


E       ╔══════════════════════════════════════════════════════╗
E       ║ Host system is missing dependencies to run browsers. ║
E       ║ Missing libraries:                                   ║
E       ║     libgobject-2.0.so.0                              ║
E       ║     libglib-2.0.so.0                                 ║
E       ║     libnss3.so                                       ║
E       ║     libnssutil3.so                                   ║
E       ║     libsmime3.so                                     ║
E       ║     libnspr4.so                                      ║
E       ║     libatk-1.0.so.0                                  ║
E       ║     libatk-bridge-2.0.so.0                           ║
E       ║     libcups.so.2                                     ║
E       ║     libgio-2.0.so.0                                  ║
E       ║     libdrm.so.2                                      ║
E       ║     libdbus-1.so.3                                   ║
E       ║     libxcb.so.1                                      ║
E       ║     libxkbcommon.so.0                                ║
E       ║     libatspi.so.0                                    ║
E       ║     libX11.so.6                                      ║
E       ║     libXcomposite.so.1                               ║
E       ║     libXdamage.so.1                                  ║
E       ║     libXext.so.6                                     ║
E       ║     libXfixes.so.3                                   ║
E       ║     libXrandr.so.2                                   ║
E       ║     libgbm.so.1                                      ║
E       ║     libpango-1.0.so.0                                ║
E       ║     libcairo.so.2                                    ║
E       ║     libasound.so.2                                   ║
E       ╚══════════════════════════════════════════════════════╝

```


**Expected behavior**
Don't receive failure

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -64,8 +64,10 @@ steps:
       firebase deploy --project $PROJECT_ID --only hosting
 
   # Optionally run tests, confirming deployment (run with '--substitutions=_RUN_TESTS=yes')
+  # Use Debian 11 Bullseye until Playwright supports Debian 12 Bookworm
+  #   - https://playwright.dev/python/docs/intro#system-requirements
   - id: "test deployment"
-    name: python:3.11-slim
+    name: python:3.11-slim-bullseye
     dir: provisioning
     env:
       - _RUN_TESTS=$_RUN_TESTS
@@ -78,8 +80,7 @@ steps:
       fi
 
       python -m pip install -r test/requirements.txt
-      playwright install-deps
-      playwright install
+      playwright install --with-deps
       python -m pytest
 
 timeout: 1800s

--- a/provisioning/automation/cloudbuild.yaml
+++ b/provisioning/automation/cloudbuild.yaml
@@ -24,8 +24,11 @@ steps:
         "--project=$_CI_PROJECT"
       ]
 
+
+  # Use Debian 11 Bullseye until Playwright supports Debian 12 Bookworm
+  #   - https://playwright.dev/python/docs/intro#system-requirements
   - id: "test"
-    name: python:3.11-slim
+    name: python:3.11-slim-bullseye
     dir: provisioning
     env: 
       - 'CI_PROJECT=$_CI_PROJECT'

--- a/provisioning/automation/cloudbuild.yaml
+++ b/provisioning/automation/cloudbuild.yaml
@@ -36,8 +36,7 @@ steps:
         #!/bin/bash
 
         python -m pip install -r test/requirements.txt
-        playwright install-deps
-        playwright install
+        playwright install --with-deps
         python -m pytest
 
 

--- a/provisioning/test/README.md
+++ b/provisioning/test/README.md
@@ -17,8 +17,7 @@ Install the dependencies
 
 ```
 python -m pip install -r test/requirements.txt
-playwright install-deps  
-playwright install
+playwright install --with-deps
 ```
 
 Then run the tests: 

--- a/provisioning/test/test.cloudbuild.yaml
+++ b/provisioning/test/test.cloudbuild.yaml
@@ -1,8 +1,11 @@
 steps:
 
   # Only run tests against the current project
+  # Use latest Python version
+  # Use Debian 11 Bullseye until Playwright supports Debian 12 Bookworm
+  #   - https://playwright.dev/python/docs/intro#system-requirements
   - id: "test"
-    name: python:3.10-slim
+    name: python:3.11-slim-bullseye
     env:
       - AVOCANO_PURCHASE_MODE=$_PURCHASE_MODE
     dir: provisioning

--- a/provisioning/test/test.cloudbuild.yaml
+++ b/provisioning/test/test.cloudbuild.yaml
@@ -12,8 +12,7 @@ steps:
     script: |
         #!/bin/bash
         python -m pip install -r test/requirements.txt
-        playwright install-deps  
-        playwright install
+        playwright install --with-deps
         python -m pytest
 
 substitutions:

